<root>
    <TreeNodesModel>
        <Action ID="AssemblyMount">
            <input_port type="float" name="F_max"/>
            <input_port type="float" name="M_max"/>
            <input_port type="float" name="a_max"/>
            <input_port type="std::string" name="end_pos"/>
            <input_port type="std::string" name="item_id"/>
            <input_port type="int" name="task_id"/>
            <input_port type="float" name="v_max"/>
        </Action>
        <Action ID="AssemblyPickplace">
            <input_port type="float" name="F_max"/>
            <input_port type="float" name="M_max"/>
            <input_port type="float" name="a_max"/>
            <input_port type="std::string" name="end_pos"/>
            <input_port type="std::string" name="item_id"/>
            <input_port type="int" name="task_id"/>
            <input_port type="float" name="v_max"/>
        </Action>
        <Action ID="AssemblyPlug">
            <input_port type="float" name="F_max"/>
            <input_port type="float" name="M_max"/>
            <input_port type="float" name="a_max"/>
            <input_port type="std::string" name="end_pos"/>
            <input_port type="std::string" name="item_id"/>
            <input_port type="int" name="task_id"/>
            <input_port type="float" name="v_max"/>
        </Action>
        <Action ID="AssemblyScrew">
            <input_port type="float" name="F_max"/>
            <input_port type="float" name="M_max"/>
            <input_port type="float" name="a_max"/>
            <input_port type="std::string" name="item_id"/>
            <input_port type="std::string" name="screw_id"/>
            <input_port type="int" name="task_id"/>
            <input_port type="float" name="v_max"/>
        </Action>
        <Control ID="FixedModule"/>
        <Action ID="Grasp">
            <input_port type="float" name="F_max"/>
            <input_port type="float" name="a_max"/>
            <input_port type="std::string" name="item_id"/>
            <input_port type="int" name="task_id"/>
            <input_port type="float" name="v_max"/>
        </Action>
        <Action ID="Move">
            <input_port type="float" name="a_max"/>
            <input_port type="std::string" name="end_pos"/>
            <input_port type="std::string" name="item_id"/>
            <input_port type="int" name="task_id"/>
            <input_port type="float" name="v_max"/>
        </Action>
        <Action ID="Passform">
            <input_port type="unsigned int" name="action_type"/>
        </Action>
        <Action ID="Position">
            <input_port type="float" name="F_max"/>
            <input_port type="float" name="M_max"/>
            <input_port type="float" name="a_max"/>
            <input_port type="std::string" name="end_pos"/>
            <input_port type="std::string" name="item_id"/>
            <input_port type="int" name="task_id"/>
            <input_port type="float" name="v_max"/>
        </Action>
        <Action ID="Pressure">
            <input_port type="float" name="F_max"/>
            <input_port type="float" name="M_max"/>
            <input_port type="std::string" name="item_id"/>
            <input_port type="int" name="task_id"/>
        </Action>
        <Action ID="PrintString">
            <input_port type="std::string" name="message"/>
        </Action>
        <Decorator ID="RangeLoop">
            <output_port default="current_iteration" type="int" name="current"/>
            <input_port type="int" name="start">First value of the iteration</input_port>
            <input_port default="1" type="int" name="step">The increment at each iteration</input_port>
            <input_port type="int" name="stop">Iteration returns before last value</input_port>
        </Decorator>
        <Action ID="Release">
            <input_port type="int" name="task_id"/>
        </Action>
        <Control ID="SpecificModule">
            <input_port type="std::string" name="module_uuid"/>
        </Control>
        <Action ID="ToolGet">
            <input_port type="std::string" name="item_id"/>
            <input_port type="std::string" name="task_id"/>
            <input_port type="std::string" name="tool_id"/>
        </Action>
        <Action ID="ToolPut">
            <input_port type="std::string" name="end_pos"/>
            <input_port type="int" name="task_id"/>
            <input_port type="std::string" name="tool_id"/>
        </Action>
        <Action ID="ToolUse">
            <input_port type="std::string" name="item_id"/>
            <input_port type="std::string" name="task_id"/>
            <input_port type="std::string" name="tool_id"/>
        </Action>
    </TreeNodesModel>
</root>

